src/routes/CreateNewSafePage/steps/OwnersAndConfirmationsNewSafeStep.tsx

```diff
@@ -0,0 +1,305 @@
+              Learn about which Safe setup to use
+            </Text>
+            <Icon size="sm" type="externalLink" color="primary" />
+          </Link>
+          . The new Safe will ONLY be available on <NetworkLabel />
+        </Paragraph>
+      </Block>
+      <Hairline />
+      <Row className={classes.header}>
+        <Col xs={3}>NAME</Col>
+        <Col xs={7}>ADDRESS</Col>
+      </Row>
+      <Hairline />
+      <Block margin="md" padding="md">
+        <Row className={classes.header}>
+          {owners.map(({ nameFieldName, addressFieldName }) => {
+            const hasOwnerAddressError = formErrors[addressFieldName]
+            const showDeleteIcon = addressFieldName !== 'owner-address-0' // we hide de delete icon for the first owner
+
+            const handleScan = (address: string, closeQrModal: () => void): void => {
+              createSafeForm.change(addressFieldName, address)
+              closeQrModal()
+            }
+
+            return (
+              <Fragment key={addressFieldName}>
+                <Col xs={3}>
+                  <Field
+                    className={classes.ownerNameField}
+                    component={TextField}
+                    name={nameFieldName}
+                    placeholder="Owner Name"
+                    text="Owner Name"
+                    type="text"
+                    validate={minMaxLength(0, 50)}
+                    testId={nameFieldName}
+                  />
+                </Col>
+                <Col xs={7}>
+                  <AddressInput
+                    fieldMutator={(address) => {
+                      createSafeForm.change(addressFieldName, address)
+                      const addressName = addressBook[address]?.name
+                      if (addressName) {
+                        createSafeForm.change(nameFieldName, addressName)
+                      }
+                    }}
+                    // eslint-disable-next-line
+                    // @ts-ignore
+                    inputAdornment={
+                      !hasOwnerAddressError && {
+                        endAdornment: (
+                          <InputAdornment position="end">
+                            <CheckCircle
+                              className={classes.checkIcon}
+                              data-testid={`${addressFieldName}-valid-adornment`}
+                            />
+                          </InputAdornment>
+                        ),
+                      }
+                    }
+                    name={addressFieldName}
+                    required
+                    placeholder="Owner Address*"
+                    text="Owner Address"
+                    testId={addressFieldName}
+                  />
+                </Col>
+                <Col xs={1} center="xs" className={classes.ownerIcons} middle="xs">
+                  <ScanQRWrapper handleScan={handleScan} testId={`${addressFieldName}-scan-QR`} />
```

Comment:
In relation to my previous comment about `testId`. Maybe you can select it via a method which represents the user interaction more.